src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -661,15 +667,19 @@ public String getDatabase() {
   }
 
   public void setDatabase(String database) {
-    this.database = database;
+    if (!Strings.isNullOrEmpty(database)) {
```

Comment:
These functions (setDatabase, setSchema, setWarehouse) are called with each REsultSet creation to update the session's db/session/wh. Now, only update these values in the session if the values are non-null. Otherwise assume these values stayed the same.